File: frontend/src/types/match.ts
export type TeamSummary = {
  id: number;
  code: string;
  name: string;
};

export type Match = {
  id: number;
  match_code: string;
  date: string;
  time_utc: string | null;
  stage: string;
  group_code: string | null;
  team_a: TeamSummary;
  team_b: TeamSummary;
  venue: string | null;
  city: string | null;
  status: "scheduled" | "live" | "finished" | "postponed";
  score_a: number | null;
  score_b: number | null;
};

File: backend/app/schemas/match.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class TeamSummary(BaseModel):
    id: int
    code: str
    name: str

    model_config = ConfigDict(from_attributes=True)


class MatchListItem(BaseModel):
    id: int
    match_code: str
    date: date
    time_utc: datetime | None
    stage: str
    group_code: str | None
    team_a: TeamSummary
    team_b: TeamSummary
    venue: str | None
    city: str | None
    status: str
    score_a: int | None
    score_b: int | None

    model_config = ConfigDict(from_attributes=True)


class MatchDetail(MatchListItem):
    pass

